testutils/clustermanager/gke.go

```diff
@@ -314,38 +299,56 @@ func (gc *GKECluster) ensureProtected() {
 	}
 }
 
-// checks for existing cluster by looking at kubeconfig,
-// and sets up gc.Project and gc.Cluster properly, otherwise fail it.
-// if project can be derived from gcloud, sets it up as well
+// checkEnvironment checks environment set for kubeconfig and glcoud, and try to
+// idefity existing project/cluster if they are not set
+//
+// checks for existing cluster by looking at kubeconfig, if kubeconfig is set:
+// 	- If it exists in GKE:
+//		- If Request doesn't contain project/clustername:
+//			- Use it
+//		- If Request contains any of project/clustername:
+//			- If the cluster matches with them:
+//				- Use it
+// If cluster isn't discovered above, try to get project from gcloud
 func (gc *GKECluster) checkEnvironment() error {
-	var err error
-	// if kubeconfig is configured, use it
 	output, err := common.StandardExec("kubectl", "config", "current-context")
+	// if kubeconfig is configured, try to use it
 	if err == nil {
 		currentContext := strings.TrimSpace(string(output))
 		if strings.HasPrefix(currentContext, "gke_") {
 			// output should be in the form of gke_PROJECT_REGION_CLUSTER
 			parts := strings.Split(currentContext, "_")
 			if len(parts) != 4 { // fall through with warning
-				log.Printf("WARNING: ignoring kubectl current-context since it's malformed: '%s'", currentContext)
+				log.Printf("WARNING: ignoring kubectl current-context since it's malformed: %q", currentContext)
 			} else {
-				log.Printf("kubeconfig isn't empty, uses this cluster for running tests: %s", currentContext)
-				gc.Project = &parts[1]
+				log.Printf("kubeconfig is: %q", currentContext)
```

Comment:
This should go after line 317, it's useful for debugging.

Reply:
done